### ISSUE TITLE
xyz2grd and grd2xyz handle nans differently

### ISSUE THREAD

**Issue by kmaterna** [GenericMappingTools/gmt#4572]
I am trying to write data to and from grd files. As a test, I took a grd file to asciii and then back again (grd2xyz -> xyz2grd).  It changed the way nans were represented.  I expected that the string 'NaN' in ascii, as produced by grd2xyz, could be parsed as nans by xyz2grd, but apparently that's not the case.  It would be helpful if these two functions could do the reverse of one another.  

```
# Code for reading and writing
datafile='test_data.grd'
incvar=`gmt grdinfo -I test_data.grd`
rangevar=`gmt grdinfo -I- test_data.grd`

# Desired script: Let GMT handle nans and missing values (DOESN'T WORK)
# Similar thing happens if you put -diNaN and -doNaN in the first and second call respectively
gmt grd2xyz $datafile -ZBLa -Vd > test.xyz   # writes many 'NaN' into xyz file
gmt xyz2grd test.xyz -Goutput.grd $incvar $rangevar -ZBLa -r -fg -Vd  # reads xyz file but can't handle 'NaN'

# Workaround: provide a number [-9999] for missing values (WORKS)
gmt grd2xyz $datafile -ZBLa -Vd -do-9999 > test.xyz   # writes many '-9999' into xyz file
gmt xyz2grd test.xyz -Goutput.grd $incvar $rangevar -ZBLa -r -fg -Vd -di-9999   # works! 


# Code for plotting: 
range="-118.5/-113.5/31.7/36"
proj="M4.5i"
output="two_plots.ps"
gmt makecpt -T0/1/0.01 -Cgray > mycpt.cpt

gmt pscoast -R$range -J$proj -Wthin,black -Sgray -Di -B1.0 -K > $output
gmt grdimage test_data.grd -R$range -J$proj -Cmycpt.cpt -Q -O -K >> $output

gmt pscoast -R$range -J$proj -Wthin,black -Sgray -Di -B1.0 -X5.5i -Y0 -O -K >> $output
gmt grdimage output.grd -R$range -J$proj -Cmycpt.cpt -Q -O -K >> $output
```

No error was raised with -Vd

**Actual outcome**

<img width="1455" alt="not_working" src="https://user-images.githubusercontent.com/12519252/102125980-4f38a800-3dff-11eb-8394-1d08a59c9565.png">

**Expected outcome**

<img width="1431" alt="working" src="https://user-images.githubusercontent.com/12519252/102125991-55c71f80-3dff-11eb-9d5d-b209484adb3b.png">

**System information**
* Operating system: Mac Mojave 10.14.5
* GMT version (`gmt --version`): 6.0.0

Data I ran this on:
https://drive.google.com/drive/folders/1gjQPXDRpsB1CL-AFaVwNPkdVsJ4p4iX8

I realize that this may be a consequence of GMT's intended behavior re nans vs missing values, but it's misleading when the output of one of these functions cannot be read by the input of the other.

**Comment by PaulWessel** [comment]
Thanks. I am looking at this and can reproduce with a simple small file.  I will fix this.  However, may I ask what you are trying to achieve in those steps?  If the -R -I -r are not "correct" then this would give junk and possibly aliasing.  What is the problem you are trying to solve?

**Comment by kmaterna** [comment]
Hi Paul!  I often operate on netcdf files in Python, and I'm trying to write the outputs from Python into valid, pixel-registered grd files.  Long story short, I think my best option is to dump the output into a table and then call xyz2grd with the associated range and the proper flags for pixel-node registration, so GMT can format the files as it needs.  I encountered this bug while testing that process.  

Longer version: Python's netcdf libraries don't have a concept of pixel-node vs gridline-node registration. Reading pixel-node registered netcdf files into Python apparently loses that information.  Then, when you write a netcdf file from Python, the resulting registration seems to depend in horrible ways on your operating system, the age of your computer, and your choice of float32 vs float64 (see : https://forum.generic-mapping-tools.org/t/writing-netcdf-from-python-changes-pixel-gridline-node-registration/954).   It doesn't seem to matter which netcdf library I'm using.  So, the pathway through xyz2grd is the best I've come up with so far.  Let me know if you need more information!

**Comment by PaulWessel** [comment]
Thanks, yes, I have heard about the registration insanity.  I am curious how long it will take for "them" to realize the insanity and introduce registration options.  Science is full of these two types of grids.

OK, sounds like you know what you are doing; I was just concerned with the **-r** flag thrown in etc.  The bug you identified has been fixed and the PR #4578 is awaiting approval by the team - usually pretty quickly, and them you need to update via master.  Or for now, use your workaround.  If you were able to do the "grd2xyz" part from Python to a binary file (e.g., floats) then -**ZBLf** would not be affected and work much faster for larger files.

**Comment by kmaterna** [comment]
Thank you for the fix and for the helpful information!  This is all very good to know.
